pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -84,6 +84,11 @@ type ipSecKey struct {
 	Aead  *netlink.XfrmStateAlgo
```

Comment:
> This patch provides an elegant solution at low cost: if cilium-agent wants to remove a xfrm state catching specific mark, it has to temporarily remove the xfrm state catching general mark first and add it back after:

An alternative here would have been to rely on `ip xfrm state deleteall` to delete all that matches then re-add the catch-all state. I might have been easier to implement and a little faster. Unfortunately, our netlink library doesn't seem to support `deleteall` so it's probably not worth it :disappointed: 

I also checked if we could match on the reqid to make the deletion more precise but that's not support in `ip x s delete`.

Reply:
Thank you Paul! This is inspiring information that we may further improve in the future. I'll resolve this for now to unblock merging.